### PR TITLE
Create label column, squash description

### PR DESCRIPTION
**Pull request by kaapstorm**
## Product Description

This is a small change that changes the appearance of the Data Dictionary page.

The "Label" field moves inline with the row, and the "Description" field gets a uniform height.

#### Before
image

#### After
image

## Technical Summary

Context: [SC-3699](https://dimagi.atlassian.net/browse/SC-3699)

### Safety story

Tested locally.

### Automated test coverage

UI change. Not covered by tests.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[SC-3699]: https://dimagi.atlassian.net/browse/SC-3699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html [hunk 1]
```diff
@@ -200,6 +201,7 @@ <h3 data-bind="text: $root.activeCaseType()" style="display: inline-block;"></h3
           <div class="table-row table-header">
           <div class="row-item-small"></div>
           <div class="row-item">{% trans "Case Property" %}</div>
+          <div class="row-item">{% trans "Label" %}</div>
```
Comment (zandre-eng) [inline corehq/apps/data_dictionary/templates/data_dictionary/base.html:204]: You might just have to add an extra column for the case property group rows. Looking at the before/after screenshots, the "Label" heading is now stretching over the group name input a little bit and the "Case Property Group" text on the group row is a little off from the "Data Type" heading.

**Review by ajeety4 — COMMENTED** [review]
Thanks for this improvement.
Looks much better. +1 to Zandre's suggestion.

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html (hunk 1 above)
Comment (kaapstorm) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:204]: cbaecc7c21e

**Review by zandre-eng — APPROVED** [review]
Looks good from my end!

**Review by ajeety4 — APPROVED** [review]

**Pull request merged**